contentcuration/contentcuration/tests/test_contentnodes.py

```diff
@@ -43,22 +39,70 @@ def _create_nodes(num_nodes, title, parent=None, levels=2):
         assert node.parent == parent
         if original_channel_id:
-            assert node.original_channel_id == original_channel_id,\
-                "Node {} with title {} has an incorrect original_channel_id.".\
-                format(node.pk, node.title)
+            assert (
+                node.original_channel_id == original_channel_id
+            ), "Node {} with title {} has an incorrect original_channel_id.".format(
+                node.pk, node.title
+            )
         if channel:
             assert node.get_channel() == channel
         if source_channel_id:
             assert node.source_channel_id == source_channel_id
         _check_nodes(node, title, original_channel_id, source_channel_id, channel)
 
 
+def _check_files_for_object(source, copy):
+    source_files = source.files.all().order_by("file_on_disk")
+    copy_files = copy.files.all().order_by("file_on_disk")
+    assert len(source_files) == len(copy_files)
+    for source_file, copy_file in zip(source_files, copy_files):
+        assert source_file.file_on_disk == copy_file.file_on_disk
+        assert source_file.preset_id == copy_file.preset_id
+
+
+def _check_node_copy(source, copy, original_channel_id=None, channel=None):
+    source_children = source.get_children()
+    copy.refresh_from_db()
+    copy_children = copy.get_children()
+    assert len(source_children) == len(copy_children)
+    for child_source, child_copy in zip(source_children, copy_children):
+        assert child_copy.title == child_source.title
+        assert child_copy.parent == copy
+        assert child_copy.original_channel_id == (
+            child_source.original_channel_id or original_channel_id
+        ), "Node {} with title {} has an incorrect original_channel_id.".format(
+            child_copy.pk, child_copy.title
+        )
+        assert (
+            child_copy.original_source_node_id == source.original_source_node_id
+            or source.node_id
+        )
+        if channel:
+            assert child_copy.get_channel() == channel
+        assert child_copy.source_channel_id == child_source.get_channel().id
+        _check_files_for_object(child_source, child_copy)
+        source_assessments = child_source.assessment_items.all().order_by("order")
+        copy_assessments = child_copy.assessment_items.all().order_by("order")
+        assert len(source_assessments) == len(copy_assessments)
+        for source_assessment, copy_assessment in zip(
+            source_assessments, copy_assessments
+        ):
+            _check_files_for_object(source_assessment, copy_assessment)
+            assert source_assessment.question == copy_assessment.question
+            assert source_assessment.answers == copy_assessment.answers
+            assert source_assessment.hints == copy_assessment.hints
+            assert source_assessment.assessment_id == copy_assessment.assessment_id
+        _check_node_copy(child_source, child_copy, original_channel_id, channel)
```

Comment:
A couple other potentially useful tests:
* Do tags get copied over?
* Do prerequisites/postrequisites not get set?
* Is the node_id changed? Is content_id the same?
* If we want to handle freeze_authoring_data more granularly, does it get set as expected?
* Is changed set to True and published set to False?
* Does complete match whether or not the original node was complete?

Reply:
Thanks - will add!

Reply:
One thought - should we copy prerequisites for nodes that get copied together? Thinking if we do end up adding the KA prerequisites on the KA channel, then copying them might desirable.

Reply:
Have added all of the above - will leave the prerequisite copying idea as a potential future enhancement.